packages/mui-joy/src/Button/Button.tsx

```diff
@@ -255,6 +257,26 @@ const Button = React.forwardRef(function Button(inProps, ref) {
   };
 
   const classes = useUtilityClasses(ownerState);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (toggleButtonGroup) {
+      props.onClick?.(event);
+      toggleButtonGroup.onClick(event, props.value);
+    } else {
+      props.onClick?.(event);
+    }
+  };
+
+  const isAriaPressed = () => {
+    if (toggleButtonGroup) {
+      if (Array.isArray(toggleButtonGroup.value)) {
+        return toggleButtonGroup.value.indexOf(props.value as string | number) !== -1;
+      }
+      return toggleButtonGroup.value === props.value;
+    }
+    return props['aria-pressed'];
+  };
```

Comment:
I think we can simplify this to:
```suggestion
  let ariaPressed = props['aria-pressed'];
  if (toggleButtonGroup) {
    if (Array.isArray(toggleButtonGroup.value)) {
      ariaPressed = toggleButtonGroup.value.indexOf(props.value) !== -1;
    } else {
      ariaPressed = toggleButtonGroup.value === props.value;
    }
  }
```
There is no need of function since there is no parameter passed to `isAriaPressed`.